### DISCUSSION TITLE
Checkout specific submodule reference

### DISCUSSION THREAD

**Discussion by arihannu** [actions/checkout#449] [Q&A] [answered]
I have a repo with some submodules. I check it out with:
```
- uses: actions/checkout@v2
  with:
    submodules: recursive
```
However, after that, I want to checkout a specific ref for one of the submodules. How would I do that?

**Comment by aormsby** [comment, marked as answer]
Something like this should do the trick.

```shell
git fetch --depth=1 origin "refs/heads/<your ref name>"
git checkout <your ref name>
```

**Reply by arihannu** [reply]
Yes this is what we ended up doing.
